### PR TITLE
[BACKLOG-40474] JDK17 - Build issue for https://github.com/pentaho/pentaho-kettle.git kettle-ui

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

File: ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java [hunk 1]
```diff
@@ -249,10 +249,15 @@ protected String[] getParamters() {
...
+  @VisibleForTesting
+  protected JobEntryJob newJobEntryJob() {
+    return new JobEntryJob();
+  }
+
   protected void getParameters( JobMeta inputJobMeta ) {
     try {
       if ( inputJobMeta == null ) {
-        JobEntryJob jej = new JobEntryJob();
+        JobEntryJob jej = newJobEntryJob();
```
Comment (peterrinehart) [inline ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java:260]: This is fine and probably a bit easier to understand, but mockito can now mock constructors to help work around issues like this also, just for future reference.

File: ui/src/test/java/org/pentaho/di/ui/core/widget/ControlSpaceKeyAdapterTest.java [hunk 2]
```diff
@@ -25,28 +25,21 @@
...
-@RunWith( PowerMockRunner.class )
-@PowerMockIgnore( "jdk.internal.reflect.*" )
-@PrepareForTest( { PropsUI.class } )
 public class ControlSpaceKeyAdapterTest {
 
   @Before
   public void before() {
     PropsUI propsUI = mock( PropsUI.class );
-    PowerMockito.mockStatic( PropsUI.class );
+    mockStatic( PropsUI.class );
```
[the remark this answers is not in the document]
Comment (peterrinehart) [reply ui/src/test/java/org/pentaho/di/ui/core/widget/ControlSpaceKeyAdapterTest.java:42]: https://www.baeldung.com/mockito-mock-static-methods

File: ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java (hunk 1 above)
Comment (e-cuellar) [reply ui/src/main/java/org/pentaho/di/ui/job/entries/job/JobEntryJobDialog.java:260]: Yeah, I tried to solve this through mocking the constructors and didn't work in this case.

File: ui/src/test/java/org/pentaho/di/ui/repository/repositoryexplorer/controllers/BrowseControllerTest.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ public void folderWithSingleDotThrowsException() throws Exception {
     when( document.createElement( PROMPTBOX ) ).thenReturn( prompt );
 
     doNothing().when( controller ).confirm( anyString(), anyString() );
-    doReturn( prompt ).when( controller ).promptForName( any( UIRepositoryObject.class ) );
+    doReturn( prompt ).when( controller ).promptForName( any() );
```
Comment (peterrinehart) [inline ui/src/test/java/org/pentaho/di/ui/repository/repositoryexplorer/controllers/BrowseControllerTest.java:151]: FYI you can also use nullable( UIRepositoryObject.class ) here if you want to be more specific.  Then again, you wouldn't have needed to do this in the first place if we weren't handing it a null anyway ;)

**Review by peterrinehart — CHANGES_REQUESTED** [review]

File: ui/src/test/java/org/pentaho/di/ui/core/widget/ControlSpaceKeyAdapterTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (e-cuellar) [reply ui/src/test/java/org/pentaho/di/ui/core/widget/ControlSpaceKeyAdapterTest.java]: Done!

File: ui/src/test/java/org/pentaho/di/ui/repository/repositoryexplorer/controllers/BrowseControllerTest.java (hunk 3 above)
Comment (e-cuellar) [reply ui/src/test/java/org/pentaho/di/ui/repository/repositoryexplorer/controllers/BrowseControllerTest.java]: Done!

**Pull request merged**